### PR TITLE
Add Interest Writeup in DG

### PR DESCRIPTION
**Pull request by sltsheryl**
Ready for review 🙂

### CODE REVIEW

File: docs/diagrams/AddInterestRefSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+activate AddInterestCommand
+
+activate Model
+
+AddInterestCommand -> Model: getFilteredPersonList()
+Model --> AddInterestCommand: ObservableList<Person>
+deactivate Model
+activate Person
+AddInterestCommand -> Person: get(1)
+Person --> AddInterestCommand: personToEdit
```
Comment (carriezhengjr) [inline docs/diagrams/AddInterestRefSequenceDiagram.puml:23]: Should the return arrow be at the end of the activation bar for `personToEdit` 🤔 ?
Refer to: https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/uml.html#sequence-diagrams "[Common notation error] Activation bar too long".

File: docs/diagrams/AddInterestSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+AddInterestCommandParser -[hidden]-> AddressBookParser
+destroy AddInterestCommandParser
+
+AddressBookParser --> LogicManager
+deactivate AddressBookParser
+
+LogicManager -> AddInterestCommand : execute()
+activate AddInterestCommand
+
+ref over AddInterestCommand, Model : Add Interest
```
Comment (carriezhengjr) [inline docs/diagrams/AddInterestSequenceDiagram.puml:59]: Not sure if `sd Add Interest` diagram is required for `ref Add Interest`. 💭 
Refer to: https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/uml.html#reference-frames "Reference Frames".

**Review by carriezhengjr — COMMENTED** [review]
Some areas that you could double check with the textbook. 😃

File: docs/diagrams/AddInterestRefSequenceDiagram.puml (hunk 1 above)
Comment (sltsheryl) [reply docs/diagrams/AddInterestRefSequenceDiagram.puml:24]: Yup this is an error I made. I made several more errors which I changed

File: docs/diagrams/AddInterestSequenceDiagram.puml (hunk 2 above)
Comment (sltsheryl) [reply docs/diagrams/AddInterestSequenceDiagram.puml:58]: Yes added sd

File: docs/diagrams/AddInterestSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+activate AddInterestCommand
+
+AddInterestCommand --> AddInterestCommandParser
+deactivate AddInterestCommand
+
+AddInterestCommandParser --> AddressBookParser
+deactivate AddInterestCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+AddInterestCommandParser -[hidden]-> AddressBookParser
+destroy AddInterestCommandParser
```
Comment (carriezhengjr) [inline docs/diagrams/AddInterestSequenceDiagram.puml:50]: I think the lifeline should not continue after the X.
UML uses an X at the __end of the lifeline__ of an object to show its deletion.
Refer to Sequence diagram --> Object Deletion: https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/uml.html#object-deletion

**Review by carriezhengjr — COMMENTED** [review]
A notation that you may want to check it out. 🤔

File: docs/diagrams/AddInterestSequenceDiagram.puml (hunk 3 above)
Comment (sltsheryl) [reply docs/diagrams/AddInterestSequenceDiagram.puml:50]: PlantUML shortcomings :(
https://www.google.com/search?q=plantuml+end+lifeline+when+destroyed&oq=plantuml+end+lifeline+when+destroyed&aqs=chrome..69i57j33i160.9938j0j9&sourceid=chrome&ie=UTF-8

**Review by carriezhengjr — APPROVED** [review]
LGTM. 
You may want to ask about the "X" and continuing lifeline in forum. 🤔

**Pull request merged**